### PR TITLE
Page Object Model applied

### PR DESCRIPTION
**Pull request by DanielCalleCO**

### CODE REVIEW

File: cypress/e2e/buy-shirt.cy.ts [hunk 1]
```diff
@@ -1,24 +1,25 @@
...
-        "#center_column a.button.ajax_add_to_cart_button.btn.btn-default",
-    ).click();
-    cy.get("[style*='display: block;'] .button-container > a").click();
-    cy.get(".cart_navigation span").click();
-
-    cy.get("#email").type("aperdomobo@gmail.com");
-    cy.get("#passwd").type("WorkshopProtractor");
-    cy.get("#SubmitLogin").click();
+import {MenuContentPage, ProductListPage, ShopingCartPage, LoginPage} from "../page/index";
+import {AddressPage, ShippingPage, PaymentPage} from "../page/index";
```
Comment (holgiosalos) [inline cypress/e2e/buy-shirt.cy.ts:2]: you can rewrite this as:
```suggestion
import {
  MenuContentPage,
  ProductListPage,
  ShopingCartPage,
  LoginPage,
  AddressPage,
  ShippingPage,
  PaymentPage
} from "../page/index";
```

File: cypress/e2e/buy-shirt.cy.ts [hunk 2]
```diff
@@ -1,24 +1,25 @@
...
+    menuContentPage.goToTShirtMenu();
+    productListPage.visitProductList();
+    productListPage.goToCheckout();
+    shopingCartPage.visitShopingCart();
+    loginPage.logIn();
+    addressPage.confirmAdress();
+    shippingPage.confirmTermsAndShip();
+    paymentPage.confirmPaymentAndShip();
 
     cy.get("#center_column > div > p > strong").should(
```
Comment (holgiosalos) [inline cypress/e2e/buy-shirt.cy.ts:24]: this command, can go on the patmentPage as a method called getConfirmationTitle: `cy.get("#center_column > div > p > strong")`

File: cypress/page/login.page.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+    constructor() {
+        this.email = "#email";
+        this.passwd = "#passwd";
+        this.login = "#SubmitLogin";
+    }
+
+    public logIn(): void {
+        cy.get(this.email).type("aperdomobo@gmail.com");
+        cy.get(this.passwd).type("WorkshopProtractor");
```
Comment (holgiosalos) [inline cypress/page/login.page.ts:14]: you can receive the email and pass as parameters of the method, so in that case, this page object can be used to login any user

File: cypress/page/payment-step.page.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+        this.payment = "[ title = \"Pay by bank wire\" ]";
+        this.confirm = "I confirm my order";
+        this.button = "button";
+    }
+
+    public confirmPaymentAndShip(): void {
+
+        cy.get(this.payment).click();
+        cy.get(this.button).contains(this.confirm).click();
+    }
```
Comment (holgiosalos) [inline cypress/page/payment-step.page.ts:16]: this method has two responsibilities:

1. click on the pay by bank wire option
2. click on the confirm my order button

I would split this method in two, one for each responsibility

File: cypress/e2e/buy-shirt.cy.ts [hunk 5]
```diff
@@ -16,14 +22,11 @@ describe("Buy a t-shirt", () => {
...
     shippingPage.confirmTermsAndShip();
-    paymentPage.confirmPaymentAndShip();
-
-    cy.get("#center_column > div > p > strong").should(
-        "have.text",
-        "Your order on My Store is complete.",
-    );
+    paymentPage.confirmPayment();
+    paymentPage.shipOrder();
+    paymentPage.getConfirmationTitle();
```
Comment (holgiosalos) [inline cypress/e2e/buy-shirt.cy.ts:30]: this method, as you are calling it getConfirmationTitle, should retrieve the confirmation message, and then you can perform the assertion here, something like:
```
paymentPage.getConfirmationTitle().should(
    "have.text",
    "Your order on My Store is complete.",
);
```

File: cypress/e2e/buy-shirt.cy.ts [hunk 6]
```diff
@@ -16,14 +22,11 @@ describe("Buy a t-shirt", () => {
...
+    loginPage.logIn("aperdomobo@gmail.com", "WorkshopProtractor");
     addressPage.confirmAdress();
     shippingPage.confirmTermsAndShip();
-    paymentPage.confirmPaymentAndShip();
-
-    cy.get("#center_column > div > p > strong").should(
-        "have.text",
-        "Your order on My Store is complete.",
-    );
+    paymentPage.confirmPayment();
```
Comment (holgiosalos) [inline cypress/e2e/buy-shirt.cy.ts:28]: this is the selection of payment by bank wire, the method name should be something that tells that.

File: cypress/e2e/buy-shirt.cy.ts [hunk 7]
```diff
@@ -16,14 +22,11 @@ describe("Buy a t-shirt", () => {
...
+    paymentPage.shipOrder();
```
Comment (holgiosalos) [inline cypress/e2e/buy-shirt.cy.ts:29]: this is the click on the confirm button, the method name should reflect that: confirmOrder or something similar can be a good name

File: cypress/e2e/buy-shirt.cy.ts (hunk 5 above)
Comment (DanielCalleCO) [reply cypress/e2e/buy-shirt.cy.ts:30]: Holmes, when i tried to put .should after getConfirmationTitle( ), i got an error like: "should does not exist on type "void""... so, I decided to parameterize both values... Let me know if as i typed works...

File: cypress/e2e/buy-shirt.cy.ts [hunk 8]
```diff
@@ -1,28 +1,33 @@
...
+    menuContentPage.goToTShirtMenu();
+    productListPage.visitProductList();
+    productListPage.goToCheckout();
+    shopingCartPage.visitShopingCart();
+    loginPage.logIn("aperdomobo@gmail.com", "WorkshopProtractor");
+    addressPage.confirmAdress();
+    shippingPage.confirmTermsAndShip();
+    paymentPage.confirmBankWirePayment();
+    paymentPage.confirmOrder();
+    paymentPage.getConfirmationTitle("have.text",
```
Comment (holgiosalos) [inline cypress/e2e/buy-shirt.cy.ts:30]: I prefer to have the `should` method here not within the `getConfirmationTitle`

Comment (DanielCalleCO) [reply cypress/e2e/buy-shirt.cy.ts]: Something like as I uploaded

File: cypress/page/payment-step.page.ts [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+
+        cy.get(this.payment).click();
+    }
+    public confirmOrder(): void {
+        cy.get(this.button).contains(this.confirm).click();
+    }
+
+public getConfirmationTitle(): void {
+    cy.get(this.title);
+}
```
Comment (holgiosalos) [inline cypress/page/payment-step.page.ts:24]:
```suggestion
public getConfirmationTitle() {
    return cy.get(this.title);
}
```

**Review by holgiosalos — APPROVED** [review]

**Pull request merged**